src/server/scripts/Zandalar/Underrot/boss_elder_leaxa.cpp

```diff
@@ -0,0 +1,385 @@
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "AreaTrigger.h"
+#include "AreaTriggerAI.h"
+#include "CreatureAI.h"
+#include "CreatureAIImpl.h"
+#include "Containers.h"
+#include "InstanceScript.h"
+#include "MotionMaster.h"
+#include "ScriptMgr.h"
+#include "ScriptedCreature.h"
+#include "SpellScript.h"
+#include "SpellAuras.h"
+#include "SharedDefines.h"
+#include "TemporarySummon.h"
+#include "underrot.h"
+
+enum LeaxaSpells
+{
+    SPELL_TAINT_OF_GHUUN                = 260685,
+    SPELL_BLOOD_BOLT                    = 260879,
+    SPELL_SANGUINE_FEAST                = 264747,
+    SPELL_CREEPING_ROT_SELECTOR         = 260889,
+    SPELL_CREEPING_ROT_SUMMON           = 260894,
+    SPELL_CREEPING_ROT_SPAWN_VISUAL     = 260891,
+    SPELL_CREEPING_ROT_AURA             = 261496,
+    SPELL_CREEPING_ROT_PERIODIC_DMG     = 261498,
+    SPELL_BLOOD_MIRROR_SELECTOR         = 264603,
+    SPELL_BLOOD_MIRROR_MISSILE          = 264609
+};
+
+enum LeaxaEvents
+{
+    EVENT_BLOOD_BOLT        = 1,
+    EVENT_CREEPING_ROT,
+    EVENT_SANGUINE_FEAST,
+    EVENT_BLOOD_MIRROR
+};
+
+enum LeaxaTexts
+{
+    SAY_AGGRO                       = 0,
+    SAY_SANGUINE_FEAST              = 1,
+    SAY_ROT_AND_WITHER              = 2,
+    SAY_BLOOD_MIRROR                = 3,
+    SAY_ANNOUNCE_BLOOD_MIRROR       = 4,
+    SAY_DEATH                       = 5
+};
+
+enum LeaxaAnimKits
+{
+    BLOOD_EFFIGY_DEATH              = 9798
+};
+
+// 131318 - Elder Leaxa
+struct boss_elder_leaxa : public BossAI
+{
+    boss_elder_leaxa(Creature* creature) : BossAI(creature, DATA_ELDER_LEAXA) { }
+
+    void JustDied(Unit* /*killer*/) override
+    {
+        _JustDied();
+        instance->SendEncounterUnit(ENCOUNTER_FRAME_DISENGAGE, me);
+        Talk(SAY_DEATH);
+    }
+
+    void EnterEvadeMode(EvadeReason /*why*/) override
+    {
+        instance->SendEncounterUnit(ENCOUNTER_FRAME_DISENGAGE, me);
+
+        summons.DespawnAll();
+        _EnterEvadeMode();
+        _DespawnAtEvade();
+    }
+
+    void JustEngagedWith(Unit* who) override
+    {
+        BossAI::JustEngagedWith(who);
+        Talk(SAY_AGGRO);
+        me->SetAIAnimKitId(0);
+        events.ScheduleEvent(EVENT_BLOOD_BOLT, 1s);
+        events.ScheduleEvent(EVENT_SANGUINE_FEAST, 8s);
+        events.ScheduleEvent(EVENT_CREEPING_ROT, 12s);
+        events.ScheduleEvent(EVENT_BLOOD_MIRROR, 17s);
+        instance->SendEncounterUnit(ENCOUNTER_FRAME_ENGAGE, me, 1);
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        scheduler.Update(diff);
```

Comment:
Any reason why you‘re using the taskscheduler when you‘re not using it for anything ?

Reply:
Either he used it at some point during development and then forgot to drop it or he's got a template with a fixed struct to use by default. In any case, remove it :D